packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -2,42 +2,191 @@
+	getBindableAttributes,
+} from '../hooks/use-bindings-attributes';
+import { store as blockEditorStore } from '../store';
 import { unlock } from '../lock-unlock';
 import InspectorControls from '../components/inspector-controls';
+import BlockContext from '../components/block-context';
+
+const {
+	DropdownMenuV2: DropdownMenu,
+	DropdownMenuGroupV2: DropdownMenuGroup,
+	DropdownMenuRadioItemV2: DropdownMenuRadioItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
+	DropdownMenuItemHelpTextV2: DropdownMenuItemHelpText,
+	DropdownMenuSeparatorV2: DropdownMenuSeparator,
+} = unlock( componentsPrivateApis );
+
+const useToolsPanelDropdownMenuProps = () => {
+	const isMobile = useViewportMatch( 'medium', '<' );
+	return ! isMobile
+		? {
+				popoverProps: {
+					placement: 'left-start',
+					// For non-mobile, inner sidebar width (248px) - button width (24px) - border (1px) + padding (16px) + spacing (20px)
+					offset: 259,
+				},
+		  }
+		: {};
+};
+
+function BlockBindingsPanelDropdown( {
+	fieldsList,
+	addConnection,
+	attribute,
+	binding,
+} ) {
+	const currentKey = binding?.args?.key;
+	return (
+		<>
+			{ Object.entries( fieldsList ).map( ( [ label, fields ], i ) => (
+				<DropdownMenuGroup key={ label }>
+					{ Object.keys( fieldsList ).length > 1 && (
+						<Text
+							className="block-editor-bindings__source-label"
+							upperCase
+							variant="muted"
+						>
+							{ label }
+						</Text>
+					) }
+					{ Object.entries( fields ).map( ( [ key, value ] ) => (
+						<DropdownMenuRadioItem
+							key={ key }
+							onChange={ () => addConnection( key, attribute ) }
+							name={ attribute + '-binding' }
+							value={ key }
+							checked={ key === currentKey }
+						>
+							<DropdownMenuItemLabel numberOfLines={ 1 }>
+								{ key }
+							</DropdownMenuItemLabel>
+							<DropdownMenuItemHelpText numberOfLines={ 1 }>
+								{ value }
+							</DropdownMenuItemHelpText>
+						</DropdownMenuRadioItem>
+					) ) }
+					{ i !== Object.keys( fieldsList ).length - 1 && (
+						<DropdownMenuSeparator />
+					) }
+				</DropdownMenuGroup>
+			) ) }
+		</>
+	);
+}
+
+function BlockBindingsAttribute( { attribute, binding } ) {
+	const { source: sourceName, args } = binding || {};
+	const sourceProps =
+		unlock( blocksPrivateApis ).getBlockBindingsSource( sourceName );
+	return (
+		<VStack>
+			<Truncate>{ attribute }</Truncate>
+			{ !! binding && (
+				<Text
+					variant="muted"
+					className="block-editor-bindings__item-explanation"
+				>
+					<Truncate>
+						{ args?.key || sourceProps?.label || sourceName }
+					</Truncate>
+				</Text>
+			) }
+		</VStack>
+	);
+}
+
+function ReadOnlyBlockBindingsPanelItems( { bindings } ) {
+	return (
+		<>
+			{ Object.entries( bindings ).map( ( [ attribute, binding ] ) => (
+				<Item key={ attribute }>
+					<BlockBindingsAttribute
+						attribute={ attribute }
+						binding={ binding }
+					/>
+				</Item>
+			) ) }
+		</>
+	);
+}
```

Comment:
I agree the UI for read only needs to be revisited, and I also agree it can be done in a follow-up.

Reply:
> We could just add a gray background to the box. What do you think?
> it can be done in a follow-up.

Agreed. Also, since this part of the UI is using `ItemGroup` and `Item`, it's a good opportunity to codify what a disabled `Item` looks like with the design folks, and add support for it directly to the component.

Please ping @WordPress/gutenberg-components when you start working on it!